File: backend/app/schemas/analytics.py
"""Analytics Pydantic schemas."""
from typing import Optional, List, Dict
from datetime import date, datetime
from pydantic import BaseModel


class AnalyticsOverview(BaseModel):
    """Overview metrics for analytics dashboard."""
    total_code_generations: int
    active_users: int
    average_confidence: float
    approval_rate: float
    rejection_rate: float
    period_start: Optional[date] = None
    period_end: Optional[date] = None
    
    class Config:
        from_attributes = True


class UsageDataPoint(BaseModel):
    """Data point for usage over time chart."""
    date: date
    count: int
    brand_name: Optional[str] = None
    
    class Config:
        from_attributes = True


class QualityMetrics(BaseModel):
    """Quality metrics including confidence distribution."""
    confidence_distribution: Dict[str, int]  # ranges: 0-20%, 20-40%, etc.
    status_breakdown: Dict[str, int]  # generated, approved, rejected
    average_confidence: float
    
    class Config:
        from_attributes = True


class BrandPerformance(BaseModel):
    """Performance metrics per brand."""
    brand_id: int
    brand_name: str
    code_generations: int
    average_confidence: float
    approval_rate: float
    top_users: List[str]  # user emails
    
    class Config:
        from_attributes = True


class UserActivity(BaseModel):
    """User activity metrics."""
    user_id: int
    user_email: str
    user_name: Optional[str] = None
    code_generations: int
    average_confidence: float
    last_active: datetime
    
    class Config:
        from_attributes = True


class LLMCostMetrics(BaseModel):
    """LLM cost metrics (super admin only)."""
    total_cost_usd: float
    average_cost_per_generation: float
    total_tokens: int
    total_generations: int
    period_start: Optional[date] = None
    period_end: Optional[date] = None
    
    class Config:
        from_attributes = True


class BrandLLMCost(BaseModel):
    """LLM cost breakdown by brand (super admin only)."""
    brand_name: str
    total_cost_usd: float
    generations: int
    cost_per_generation: Optional[float] = None  # Calculated
    
    class Config:
        from_attributes = True

File: frontend/src/types/analytics.ts
/** Analytics TypeScript types matching backend schemas */

export interface AnalyticsOverview {
  total_code_generations: number
  active_users: number
  average_confidence: number
  approval_rate: number
  rejection_rate: number
  period_start?: string | null
  period_end?: string | null
}

export interface UsageDataPoint {
  date: string
  count: number
  brand_name?: string | null
}

export interface QualityMetrics {
  confidence_distribution: Record<string, number>
  status_breakdown: Record<string, number>
  average_confidence: number
}

export interface BrandPerformance {
  brand_id: number
  brand_name: string
  code_generations: number
  average_confidence: number
  approval_rate: number
  top_users: string[]
}

export interface UserActivity {
  user_id: number
  user_email: string
  user_name?: string | null
  code_generations: number
  average_confidence: number
  last_active: string
}

export interface LLMCostMetrics {
  total_cost_usd: number
  average_cost_per_generation: number
  total_tokens: number
  total_generations: number
  period_start?: string | null
  period_end?: string | null
}

export interface BrandLLMCost {
  brand_name: string
  total_cost_usd: number
  generations: number
  cost_per_generation?: number | null
}

export interface AnalyticsParams {
  start_date?: string
  end_date?: string
  brand_id?: number
  interval?: 'day' | 'week' | 'month'
  limit?: number
}

